src/cryfs-cli/program_options/ProgramOptions.cpp

```diff
@@ -16,6 +16,20 @@ ProgramOptions::ProgramOptions(const bf::path &baseDir, const bf::path &mountDir
     :_baseDir(baseDir), _mountDir(mountDir), _configFile(configFile), _foreground(foreground),
      _cipher(cipher), _blocksizeBytes(blocksizeBytes), _unmountAfterIdleMinutes(unmountAfterIdleMinutes),
      _logFile(logFile), _fuseOptions(fuseOptions) {
+
+    auto hasNoOption = [&](const char *opt) {
+        for (const string& it : _fuseOptions) {
+            if (std::strncmp(it.c_str(), opt, std::strlen(opt))) {
+                return false;
+            }
+        }
+        return true;
+    };
+
+    if (hasNoOption("subtype=cryfs") && hasNoOption("fsname=cryfs@")) {
+               _fuseOptions.push_back("-ofsname=cryfs@"+baseDir.native());
+               _fuseOptions.push_back("-osubtype=cryfs");
+    }
```

Comment:
Cool, thanks. Btw I'm usually using boost::optional and boost::none to state that something is unset. You can then use  `if (var==boost::none)` tests and dereference var like a pointer to *var to get its value once you checked it is set.

Reason for this is that empty string can't be distinguished from an explicitly set empty string and it might make sense to handle these cases separately. For example use fstype as fsname if fsname is not set, but use empty string as fsname if explicitly set.

On June 2, 2016 2:54:19 AM PDT, mhogomchungu <notifications@github.com> wrote:
>> @@ -16,6 +16,20 @@ ProgramOptions::ProgramOptions(const bf::path
>&baseDir, const bf::path &mountDir
>>      :_baseDir(baseDir), _mountDir(mountDir),
>_configFile(configFile), _foreground(foreground),
>>       _cipher(cipher), _blocksizeBytes(blocksizeBytes),
>_unmountAfterIdleMinutes(unmountAfterIdleMinutes),
>>       _logFile(logFile), _fuseOptions(fuseOptions) {
>> +
>> +    auto hasNoOption = [&](const char *opt) {
>> +        for (const string& it : _fuseOptions) {
>> +            if (std::strncmp(it.c_str(), opt, std::strlen(opt))) {
>> +                return false;
>> +            }
>> +        }
>> +        return true;
>> +    };
>> +
>> +    if (hasNoOption("subtype=cryfs") &&
>hasNoOption("fsname=cryfs@")) {
>> +              
>_fuseOptions.push_back("-ofsname=cryfs@"+baseDir.native());
>> +               _fuseOptions.push_back("-osubtype=cryfs");
>> +    }
>
>```Another option might be to pass in two string arguments to the Fuse
>constructor, one for the file system type (i.e. "-osubtype") and one
>for the file system source (i.e. "-ofsname")```
>
>My latest commit went with this option.
>
>I have "fsname" as an optional argument because the Fuse class does not
>require it.
>
>---
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/cryfs/cryfs/pull/61/files/2cf4a2e08752d444e570b59b7647d47eb00b2842..002dc6ca17e4a7b72b9ba98c5d92f89245743de1#r65512836